libdnf/dnf-repo.cpp

```diff
@@ -1729,6 +1729,7 @@ dnf_repo_update(DnfRepo *repo,
     gint64 timestamp_new = 0;
     g_autoptr(GError) error_local = NULL;
     RepoUpdateData updatedata = { 0, };
+    GCancellable *cancellable = g_cancellable_new ();
```

Comment:
`g_autoptr(GCancellable)` here.

Actually that said though, the cancellable is something that we should support the *higher level* code passing in (same as in librepo).  Ultimately for example that'd be e.g. rpm-ostree calling into libdnf.

So since it's not actually used here, we can just pass `NULL` for the cancellable for now.

But what would make sense here ultimately is to teach librepo about `GCancellable` and to use it - instead of what it's doing now in catching `SIGINT`.